### PR TITLE
Move patterns.js into lib directory

### PR DESCRIPTION
**Pull request by stefwalter**
This is not a stable base package dependency, but rather a shared development dependency.
    
Although this is ugly for now, further changes to the build system will help to bundle this in a cleaner manner.

### CODE REVIEW

**Comment by stefwalter** [comment]
Depends on #4477

**Comment by stefwalter** [comment]
Fixed two deps I broke in users and storage packages.

File: pkg/shell/machine-dialogs.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ define([
     "data!shell/templates/sync-users.html",
     "data!shell/templates/unknown-hostkey.html",
     'shell/po',
-    "base1/patterns",
+    "./patterns",
```
Comment (dperpeet) [inline pkg/shell/machine-dialogs.js]: Is there a reason you use `./` here, but the actual name in the other packages?

Comment (stefwalter) [inline pkg/shell/machine-dialogs.js]: The actual name needs to go into require() calls, but ./ is a better way to refer to local stuff from define() calls. Given the data-main support is working now we should move away from most of the require() calls.

File: pkg/storaged/dialog.js [hunk 2]
```diff
@@ -21,9 +21,9 @@ define([
     "jquery",
     "base1/cockpit",
     "base1/mustache",
-    "base1/patterns",
-    "shell/controls"
-], function($, cockpit, mustache, patterns, controls) {
+    "shell/controls",
+    "./patterns",
```
Comment (dperpeet) [inline pkg/storaged/dialog.js]: ok, so maybe you didn't use `./` only in one place

**Comment by dperpeet** [comment]
this breaks subscriptions, fix in #4543

**Comment by dperpeet** [comment]
closed in #4543

**Pull request closed without merging**